[FEA] Automate appending new operators to the platform score sheets

cindyyuanjiang opened the pull request:
Fixes https://github.com/NVIDIA/spark-rapids-tools/issues/915

**Changes**

- Add a script `sync_operator_scores.py` to automate appending new operators to the platform score sheets
- Updated `process_supported_files.py` script to generate a text file with the new operators from the plugin, this file can later be picked up by `sync_operator_scores.py`

parthosa commented on scripts/sync_plugin_files/sync_operator_scores.py:
```diff
@@ -0,0 +1,76 @@
+    with open(new_operators_file, 'r') as f:
+        for line in f:
+            new_operators.add(line.strip())
+    print(new_operators)
```
Do we need these `prints()`?

cindyyuanjiang replied:
removed, thanks!

parthosa commented on scripts/sync_plugin_files/sync_operator_scores.py:
```diff
@@ -0,0 +1,76 @@
+        for file in os.listdir(operator_score_dir):
+            file_path = os.path.join(operator_score_dir, file)
+            if "operatorsScore" in file_path and os.path.isfile(file_path):
+                print(file_path)
```
Similar to above. Do we need this print()?

cindyyuanjiang replied:
removed, thanks!

parthosa reviewed (commented):
Thanks @cindyyuanjiang. I see couple of linting issues. File length > 120, unused import. I was wondering if the python files in this directory are not checked by `tox`.

Additionally, do we need to update the README on how to use this script?

parthosa commented:
In this change, we will be using the default of `1` for all platforms/hw-configs for new operators. 

Currently we have a different default value for different platform. For example, `WindowGroupLimitExec` has value of `2.45` for aws-t4 but `4.88` for dataproc-t4. 

I wanted to know if this has any impact

cindyyuanjiang commented on scripts/sync_plugin_files/sync_operator_scores.py:
```diff
@@ -0,0 +1,76 @@
+    if os.path.exists(operator_score_dir) and os.path.isdir(operator_score_dir):
+        for file in os.listdir(operator_score_dir):
+            file_path = os.path.join(operator_score_dir, file)
+            if "operatorsScore" in file_path and os.path.isfile(file_path):
```
updated, thanks!

cindyyuanjiang commented:
> Thanks @cindyyuanjiang. I see couple of linting issues - line length > 120, unused import. I was wondering if the python files in this directory are not checked by `tox`.
> 
> Additionally, do we need to update the README on how to use this script?

Thanks @partho! Yes, this file and probably the `scripts` folder is not checked by `tox`. I have removed the unused imports and updated `README`.

cindyyuanjiang commented:
> In this change, we will be using the default of `1` for all platforms/hw-configs for new operators.

I think we are okay with a lower default score because:
1. It is conservative.
2. When we update the operator score files by running the script that generates the speedup factors, all rows will be overridden with the most up-to-date score.

parthosa reviewed (approved):
Thanks @cindyyuanjiang. LGTM.

amahussein reviewed (approved):
Thanks @cindyyuanjiang 
LGTME
